enhancements/olm/runtime-bundle-configuration.md

```diff
@@ -0,0 +1,858 @@
+| -             | ✓               | ✓            | required and can be any namespace                           |
+| ✓             | -               | -            | no configuration                                            |
+| ✓             | -               | ✓            | optional and must be install namespace (default: unset)     |
+| ✓             | ✓               | -            | optional and must NOT be install namespace (default: unset) |
+| ✓             | ✓               | ✓            | optional and can be any namespace (default: unset)          | 
+
+Given the tight scope and the small configuration surface, the validation may be done programmatically rather than through
+the actual generation of a JSON-Schema. However, once this surface expands this will be the direction we'll take and
+there should be no difference in behavior beyond the possible wording and structure of error messages.
+
+Below are examples for what the JSON-Schemas may look like:
+
+**watchNamespace field required**
+
+```
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "watchNamespace": {
+      "type": "string",
+      "format": "namespaceName",
+    }
+  },
+  "required": ["watchNamespace"]
+}
+```
+
+**watchNamespace field optional**
+
+```
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "watchNamespace": {
+      "type": ["string", null],
+      "format": "namespaceName",
+    }
+  },
+}
+```
+
+**watchNamespace cannot be the install namespace**
+
+```
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "watchNamespace": {
+      "format": "isNotInstallNamespace",
+      "type": ["string", null],
+    }
+  }
+}
+```
+
+**watchNamespace is optional and can only be the install namespace**
+
+```
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "watchNamespace": {
+      "format": "isInstallNamespace",
+      "type": ["string", null],
+    }
+  }
+}
+```
+
+**Custom JSON Schema Formats for Bundle Authors:**
+
+OLM's JSON Schema validator will support several custom formats that bundle authors can opt-in to use in their configuration schemas:
+
+- `namespaceName`: Will validate that the input is a valid Kubernetes namespace name (follows DNS-1123 subdomain format). This will be useful for any configuration field that accepts a namespace name.
+- `isInstallNamespace`: Will validate that the input is a valid namespace name and matches the ClusterExtension's install namespace (`.spec.namespace`). Bundle authors can use this format to ensure a configuration value must be the install namespace.
+- `isNotInstallNamespace`: Will validate that the input is a valid namespace name and does not match the install namespace. Bundle authors can use this format to ensure a configuration value must be different from the install namespace.
```

Comment:
Since we are generating schema for registry+v1 bundles and we will know:
- the install namespace (from `CE.spec.namespace`)
- the bundles install modes (from `CSV.spec.installModes` in the bundle)

It seems like we could forgo the custom formats for install namespace stuff and directly put the namespace names in the schema. (With `"not": {<schema>}` to disallow a namespace name).

Then we could decide later if we need to provide helpers for this functionality.